### PR TITLE
Guard against intermittent AttributeError on request.transaction

### PR DESCRIPTION
**Pull request by gbrownmozilla**
Consider, what if flask updates request.full_path between before/after? (I can't find any docs that say it will, or won't!)

Or, what if dbo.begin() raises, in setup_request()?

Otherwise, is it possible that @app.before_request is not always called??

### CODE REVIEW

**Comment by gbrownmozilla** [comment]
I don't think this will address the recent outage in version 3.15, but it should avoid the AttributeError noted in #2479 and seen intermittently in the past.

**Comment by gbrownmozilla** [comment]
I expect the logging change to produce a log message like:

`Caught ISE 500 error: AttributeError("'Request' object has no attribute 'transaction'")`

This might help us better understand ISEs.

File: src/auslib/web/admin/base.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def setup_request():
 
 @app.after_request
 def complete_request(response):
-    if request.full_path.startswith("/v2"):
+    if hasattr(request, "transaction"):
```
Comment (jcristau) [inline src/auslib/web/admin/base.py:91]: I wonder if it's worth keeping the `full_path` check, for symmetry, but maybe not.

File: src/auslib/web/admin/base.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def unicode(error):
 @app.errorhandler(Exception)
 def ise(error):
     capture_exception(error)
-    log.error("Caught ISE 500 error.")
+    log.error("Caught ISE 500 error: %r", error)
```
Comment (jcristau) [inline src/auslib/web/admin/base.py:137]: Would `log.exception` instead work?

**Review by jcristau — APPROVED** [review]

File: src/auslib/web/admin/base.py (hunk 1 above)
Comment (gbrownmozilla) [reply src/auslib/web/admin/base.py:91]: I think the hasattr way is better: We know it's possible to have "/v2" and not have request.transaction; throwing an AttributeError here doesn't help us at all.

File: src/auslib/web/admin/base.py (hunk 2 above)
Comment (gbrownmozilla) [reply src/auslib/web/admin/base.py:137]: Good idea.

**Pull request merged**